src/wayland/compositor/handlers.rs

```diff
@@ -258,6 +266,21 @@ where
                         )))
                 });
             }
+            wl_surface::Request::Offset { x, y } => {
+                PrivateSurfaceData::with_states(surface, |states| {
+                    match &mut states.cached_state.pending::<SurfaceAttributes>().buffer {
+                        Some(BufferAssignment::NewBuffer { delta, .. }) => {
+                            *delta = (x, y).into();
+                        }
+                        _ => {
+                            error!(
+                                state.compositor_state().log,
+                                "Got wl_surface::Offset request, but there is no pending buffer available."
+                            );
```

Comment:
I'd say this kind `error!` logging should be reserved to problem in the compositor (or interaction with the OS), not clients misbehaving. I'd rather have this as a `debug!` loglevel

Reply:
Done

Reply:
I don't see anything in the protocol, that requires `attach` to be called before `offset`. But because it is double-buffered, we have to handle that. I think setting an offset first, then attaching the buffer and assuming the offset will be used by the compositor after commit is perfectly legal.

Reply:
Indeed,  https://github.com/Smithay/smithay/pull/602/commits/f5752848315067239f63ee2ce749400065b960b2 should address this, it reworks the handling of the offset around the new version, and retrofits pre 5 way.